grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java

```diff
@@ -317,6 +323,28 @@ public void instanceTypeHierarchyRespected(){
         assertEquals(answers.size(), 2);
     }
 
+    @Test //Expected result: Both queries should return a single equal match as they trigger the same rule.
+    public void reasoningOverRelationHierarchy(){
+        QueryBuilder qb = testSet19.graph().graql().infer(true);
+        String queryString = "match (role1: $x, role2: $y) isa relation1;";
+        String queryString2 = "match (role1: $x, role2: $y) isa sub-relation1;";
+        QueryAnswers answers = queryAnswers(qb.parse(queryString));
+        QueryAnswers answers2 = queryAnswers(qb.parse(queryString2));
+        assertEquals(answers.size(), 1);
+        assertEquals(answers, answers2);
+    }
+
+    @Test //Expected result: Both queries should return a single equal match as they trigger the same rule.
```

Comment:
I like the comments explaining what is expected. This will help a lot next time I break reasoner ;)